command/services/export/export.go

```diff
@@ -0,0 +1,260 @@
+package export
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	serviceName    string
+	peerNames      string
+	partitionNames string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(&c.serviceName, "name", "", "(Required) Specify the name of the service you want to export.")
+	c.flags.StringVar(&c.peerNames, "consumer-peers", "", "(Required) Peers the service will be exported to, formatted as a comma-separated list. Not required for Enterprise if setting -consumer-partitions.")
+	c.flags.StringVar(&c.partitionNames, "consumer-partitions", "", "Required if not setting -consumer-peers. The local partitions within the same datacenter that the service will be exported to, formatted as a comma-separated list. Admin Partitions are a Consul Enterprise feature.")
+
+	c.http = &flags.HTTPFlags{}
+	flags.Merge(c.flags, c.http.ClientFlags())
+	flags.Merge(c.flags, c.http.MultiTenancyFlags())
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if err := c.validateFlags(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	peerNames, err := c.getPeerNames()
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	partitionNames, err := c.getPartitionNames()
```

Comment:
What happens if partitions are specified in OSS?

Reply:
This follows the same theme as other enterprise features in this PR. The command with partitions specified will go through even for OSS users, same as if you use `$ consul config write` today (since this is really all that's happening behind the scenes).

Since the established pattern is to allow Enterprise features in the OSS CLI, the correct thing to do in the longer term IMO is to add server validation for this config entry that returns an error in this case.